src/dpp/discordvoiceclient.cpp

```diff
@@ -821,6 +830,49 @@ void discord_voice_client::error(uint32_t errorcode)
 	}
 }
 
+void discord_voice_client::set_user_gain(snowflake user_id, float factor)
+{
+#ifdef HAVE_VOICE
+	int16_t gain;
+
+	if (factor < 0.0f) {
+		/* Invalid factor; must be nonnegative. */
+		return;
```

Comment:
should these invalid inputs throw dpp::exception derivatives instead? can't decide myself here.

Reply:
I am okay either way. Personally prefer more graceful error handling but throwing exceptions seems to be the common pattern in dpp.

Reply:
let's keep it as it is for now... it seems to me that programmatically specifying a negative volume simply by floating point error is a possibility and we shouldn't throw exceptions for that, in this case graceful failure is better.